src/zenml/zen_server/routers/run_templates_endpoints.py

```diff
@@ -0,0 +1,211 @@
+        make_dependable(RunTemplateFilter)
+    ),
+    hydrate: bool = False,
+    _: AuthContext = Security(authorize),
+) -> Page[RunTemplateResponse]:
+    """Get a page of run templates.
+
+    Args:
+        filter_model: Filter model used for pagination, sorting,
+            filtering.
+        hydrate: Flag deciding whether to hydrate the output model(s)
+            by including metadata fields in the response.
+
+    Returns:
+        Page of run templates.
+    """
+    return verify_permissions_and_list_entities(
+        filter_model=filter_model,
+        resource_type=ResourceType.RUN_TEMPLATE,
+        list_method=zen_store().list_run_templates,
+        hydrate=hydrate,
+    )
+
+
+@router.get(
+    "/{template_id}",
+    responses={401: error_response, 404: error_response, 422: error_response},
+)
+@handle_exceptions
+def get_run_template(
+    template_id: UUID,
+    hydrate: bool = True,
+    _: AuthContext = Security(authorize),
+) -> RunTemplateResponse:
+    """Get a run template.
+
+    Args:
+        template_id: ID of the run template to get.
+        hydrate: Flag deciding whether to hydrate the output model(s)
+            by including metadata fields in the response.
+
+    Returns:
+        The run template.
+    """
+    return verify_permissions_and_get_entity(
+        id=template_id,
+        get_method=zen_store().get_run_template,
+        hydrate=hydrate,
+    )
+
+
+@router.put(
+    "/{template_id}",
+    responses={401: error_response, 404: error_response, 422: error_response},
+)
+@handle_exceptions
+def update_run_template(
+    template_id: UUID,
+    update: RunTemplateUpdate,
+    _: AuthContext = Security(authorize),
+) -> RunTemplateResponse:
+    """Update a run template.
+
+    Args:
+        template_id: ID of the run template to get.
+        update: The updates to apply.
+
+    Returns:
+        The updated run template.
+    """
+    return verify_permissions_and_update_entity(
+        id=template_id,
+        update_model=update,
+        get_method=zen_store().get_run_template,
+        update_method=zen_store().update_run_template,
+    )
+
+
+@router.delete(
+    "/{template_id}",
+    responses={401: error_response, 404: error_response, 422: error_response},
+)
+@handle_exceptions
+def delete_run_template(
+    template_id: UUID,
+    _: AuthContext = Security(authorize),
+) -> None:
+    """Delete a run template.
+
+    Args:
+        template_id: ID of the run template to delete.
+    """
+    verify_permissions_and_delete_entity(
+        id=template_id,
+        get_method=zen_store().get_run_template,
+        delete_method=zen_store().delete_run_template,
+    )
+
+
+if server_config().workload_manager_enabled:
+
+    @router.post(
+        "/{template_id}/runs",
+        responses={
+            401: error_response,
+            404: error_response,
+            422: error_response,
+        },
+    )
+    @handle_exceptions
+    def create_template_run(
```

Comment:
```suggestion
    @router.post(
        "/{template_id}/run",
        responses={
            401: error_response,
            404: error_response,
            422: error_response,
        },
    )
    @handle_exceptions
    def run_run_template(
```

I think the naming of the endpoint and the function is a bit confusing here.

- For the endpoint, I would propose something like: `/run_templates/{template_id}/run`. Because it would execute and action instead of handling an entity.
- As for the function, I would rename it to either `run_template` or `run_run_template` which is similar to how it is defined in the `zen_store_interface`.

Reply:
I think @AlexejPenner would disagree :P

Your suggestion is not a restful endpoint that deals with resources. While yes this does execute an action, it also creates a `run` resource for a specific template. No strong opinion from my side though.

If the endpoint stays as is, I think the current name is better. Otherwise I agree with your suggestion 👍

Reply:
I wrote this thinking @AlexejPenner would be proud of me. Let's see what he thinks 😆

Reply:
I'm all for /runs - i don't like action endpoints outside of /action/...